packages/react-ui/src/widgets/ComparativeCategoryWidgetUI.js

```diff
@@ -0,0 +1,696 @@
+  key: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      color: PropTypes.string.isRequired,
+      value: PropTypes.number.isRequired
+    })
+  ).isRequired
+}).isRequired;
+
+function ComparativeCategoryUISkeleton() {
+  const classes = useStyles();
+  return (
+    <div className={classes.wrapper}>
+      <Box
+        display='flex'
+        alignItems='center'
+        justifyContent='space-between'
+        className={classes.toolbar}
+      >
+        <Typography variant='caption'>
+          <Skeleton variant='text' width={100} />
+        </Typography>
+      </Box>
+      <Box display='flex' flexDirection='column' className={classes.categoriesList}>
+        {[...Array(4)].map((_, i) => (
+          <Box key={i} py={0.5}>
+            <Box display='flex' justifyContent='space-between' flexWrap='nowrap'>
+              <Typography variant='body2' noWrap>
+                <Skeleton variant='text' width={100} />
+              </Typography>
+              <Typography variant='body2'>
+                <Skeleton variant='text' width={70} />
+              </Typography>
+            </Box>
+            {[...Array(3)].map((_, i) => (
+              <div key={i} className={classes.progressbar}></div>
+            ))}
+          </Box>
+        ))}
+      </Box>
+    </div>
+  );
+}
+
+function ComparativeCategoryTooltip({ item, names, formatter = IDENTITY_FN }) {
+  const theme = useTheme();
+  const classes = useStyles();
+
+  return (
+    <div>
+      <Typography color='inherit' variant='caption' noWrap>
+        {item.label}
+      </Typography>
+      <Box pt={1}>
+        {item.data.map((d, i) => (
+          <Box
+            key={names[i]}
+            display='flex'
+            alignItems='center'
+            gridGap={theme.spacing(0.75)}
+          >
+            <div
+              className={classes.bullet}
+              style={{
+                backgroundColor:
+                  item.key === OTHERS_KEY ? theme.palette.background.default : d.color
+              }}
+            ></div>
+            <Typography color='inherit' variant='caption'>
+              {names[i]}
+            </Typography>
+            <Box flexGrow={1}></Box>
+            <Typography color='inherit' variant='caption'>
+              {formatter(d.value)}
+            </Typography>
+          </Box>
+        ))}
+      </Box>
+    </div>
+  );
+}
+
+ComparativeCategoryTooltip.displayName = 'ComparativeCategoryTooltip';
+ComparativeCategoryTooltip.defaultProps = {
+  names: EMPTY_ARRAY,
+  formatter: IDENTITY_FN
+};
+ComparativeCategoryTooltip.propTypes = {
+  item: transposedCategoryItemPropTypes,
+  names: PropTypes.arrayOf(PropTypes.string).isRequired,
+  formatter: PropTypes.func
+};
+
+const StyledTooltip = withStyles((theme) => ({
+  tooltip: {
+    color: theme.palette.common.white,
+    maxWidth: 260
+  }
+}))(Tooltip);
+
+function CategoryItem({
```

Comment:
I would strongly suggest splitting this long file, with different components, into different files, one per component. Also I think it would probably make sense to create a bit more of structure in folders:

* widgets
    * CategoryWidgetUI
    * comparative
        * Category
            * ComparativeCategoryWidgetUI
            * CategoryItem
            * Skeleton...
            
            
            
            
            
            
---

Reply:
Allright, ComparativeCategoryWidget is now split into multiple files and in the folder `widgets/comparative`